### PR TITLE
server: close result set for FieldList

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @tiancaiamao PTAL

File: server/driver_tidb.go [hunk 1]
```diff
@@ -236,12 +236,13 @@ func (tc *TiDBContext) Auth(user *auth.UserIdentity, auth []byte, salt []byte) b
...
 // FieldList implements QueryCtx FieldList method.
-func (tc *TiDBContext) FieldList(table string) (colums []*ColumnInfo, err error) {
+func (tc *TiDBContext) FieldList(table string) (columns []*ColumnInfo, err error) {
 	rs, err := tc.Execute(goctx.Background(), "SELECT * FROM `"+table+"` LIMIT 0")
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return rs[0].Columns(), nil
+	columns = rs[0].Columns()
+	return columns, rs[0].Close()
```
Comment (shenli) [inline server/driver_tidb.go]: Should we use defer close the rs[0] after line 240?

Comment (coocood) [reply server/driver_tidb.go]: Actually, when an error is returned in `Execute`, rs should always be nil.

**Pull request merged**